### PR TITLE
update plan

### PR DESCRIPTION
**Pull request by zyndiecate**
epic https://github.com/giantswarm/giantswarm/issues/108
origin https://github.com/giantswarm/app-service/pull/518

### CODE REVIEW

**Comment by zyndiecate** [comment]
@ewoutp RFR

File: update_plan.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+		return ServiceUpdatePlan{}, mask(err)
+	}
+
+	resp, err := apischema.FromHTTPResponse(c.post(c.appUrl(apiPrefixV1, service, "/update/plan"), "application/json", bytes.NewReader(b)))
+	if err != nil {
+		return ServiceUpdatePlan{}, mask(err)
+	}
+
+	if err := resp.EnsureStatusCodes(apischema.STATUS_CODE_DATA); err != nil {
+		if apischema.IsUserError(err) {
```
Comment (ewoutp) [inline update_plan.go]: What about resource not found error?

**Comment by ewoutp** [comment]
Small question / possible error issues, otherwise LGTM

File: update_plan.go (hunk 1 above)
Comment (zyndiecate) [inline update_plan.go]: Added those checks.

**Pull request merged**